File: 프로그래머스/LV2/코딩테스트 연습/월간 코드 챌린지 시즌2/2개 이하로 다른 비트/2개 이하로 다른 비트.py
def solution(numbers):
    answer = []
    for n in numbers:
        b = 1
        temp = n
        while True:
            if temp & 1 == 0:
                break
            temp >>= 1
            b <<= 1
            
        answer.append(n + b - (b>>1))
    return answer

File: 프로그래머스/LV2/코딩테스트 연습/월간 코드 챌린지 시즌2/2개 이하로 다른 비트/2개 이하로 다른 비트.cpp
#include <string>
#include <vector>
#include <cmath>
using namespace std;

vector<long long> solution(vector<long long> numbers) 
{
    vector<long long> answer;
    long long b, temp;
    for(long long n : numbers)
    {
        b = 1;
        temp = n;
        while (true)
        {
            if(!(temp & 1))break;
            temp >>= 1;
            b <<= 1;
        }
        answer.push_back(n + b - (b>>1));  
    }
    return answer;
}